File: source/OptimaJet.DWKit.StarterApplication/Models/ProductPublication.cs
using System;
using System.Collections.Generic;
using JsonApiDotNetCore.Models;
namespace OptimaJet.DWKit.StarterApplication.Models
{
    public class ProductPublication : Identifiable, ITrackDate
    {
        [HasOne("product")]
        public virtual Product Product { get; set; }
        public Guid ProductId { get; set; }

        [HasOne("product-build")]
        public virtual ProductBuild ProductBuild { get; set; }
        public int ProductBuildId { get; set; }

        // BuildEngine Release Id
        [Attr("release-id")]
        public int ReleaseId { get; set; }

        [Attr("channel")]
        public string Channel { get; set; }

        [Attr("log-url")]
        public string LogUrl { get; set; }

        [Attr("success")]
        public bool? Success { get; set; }

        [Attr("date-created")]
        public DateTime? DateCreated { get; set; }

        [Attr("date-updated")]
        public DateTime? DateUpdated { get; set; }
    }
}

File: source/SIL.AppBuilder.Portal.Frontend/src/data/models/product-publication.ts
import { AttributesObject, ResourceObject } from 'jsonapi-typescript';

export type PRODUCT_PUBLICATION_TYPE = 'productPublication';

export interface ProductPublicationAttributes extends AttributesObject {
  channel: string;
  success?: boolean;
  releaseId: number;
  logUrl: string;
  dateCreated: string;
  dateUpdated: string;
}

export type ProductPublisheResource = ResourceObject<
  PRODUCT_PUBLICATION_TYPE,
  ProductPublicationAttributes
>;
